src/main/java/uk/ac/cam/cl/dtg/isaac/api/managers/EventBookingManager.java

```diff
@@ -181,6 +205,29 @@ public boolean isUserAbleToManageEvent(RegisteredUserDTO user, IsaacEventPageDTO
         return false;
     }
 
+    /**
+     * Check if the requesting/logged-in user is the one who made the reservation.
+     *
+     * @param user                  The logged in user
+     * @param userOwningReservation The reserved user
+     * @param event                 The event on which the reserved user is reserved
+     * @return either true or false whether the requesting user made the reservation
+     * @throws SegueDatabaseException
+     */
+    public boolean isReservationMadeByRequestingUser(RegisteredUserDTO user,
+                                                     RegisteredUserDTO userOwningReservation,
+                                                     IsaacEventPageDTO event)
+            throws SegueDatabaseException {
+
+        EventBookingDTO booking = this.getBookingByEventIdAndUserId(event.getId(), userOwningReservation.getId());
+        UserSummaryDTO reservingUser = booking.getReservedBy();
+        if (null != reservingUser) {
```

Comment:
Because this should be only used for reservations, and because reservations should always have that attribute set to not null, this null check should not be necessary, but try telling that to Java…

But yeah, I'd rather flip the `.equals()`.